desc/compute/_stability.py

```diff
@@ -0,0 +1,276 @@
+"""Compute functions for Mercier stability objectives."""
+
+from scipy.constants import mu_0
+
+from desc.backend import jnp
+from .utils import (
+    check_derivs,
+    dot,
+    compress,
+    expand,
+    surface_averages,
+    surface_integrals,
+)
+from ._core import compute_pressure, compute_toroidal_flux_gradient, compute_geometry
+from ._field import (
+    compute_magnetic_field_magnitude,
+    compute_boozer_magnetic_field,
+    compute_contravariant_current_density,
+)
+
+
+def compute_mercier_stability(
```

Comment:
1. The only drawback of removing the mercier compute functions [mercier compute functions](https://github.com/PlasmaControl/DESC/blob/c90dde34e9024ebb4b0365a607e9e414dd75662b/desc/compute/_mercier.py) for just the one here is if the user wanted `D_shear` they'd need to provide all inputs when only `iota` was needed. Pressure is only needed for DWell and DMerc too. Are we fine with this?
2. I'd prefer to add a couple local variables inside the if statements to make it more readable and avoid recomputing things. Let me know if you are opposed to this.

Reply:
1. This shouldn't be a problem in practice. The users never call the compute functions directly, so they don't have to worry about what profiles to pass in. And the tiny overhead shouldn't hurt performance. Having a separate function for each quantity resulted in a lot of duplicate code and documentation, and I wanted to reduce that
2. If something is a useful physics quantity it should be added to the `data_index`, not just used as a local variable. I was trying to make the code more closely match the equations from the paper, but I may have over done it. If you see some repeated computations that would be good to separate out